scapy/contrib/automotive/someip_sd.py

```diff
@@ -0,0 +1,342 @@
+        length = self.len
+        if (length is None):
+            length = len(self.cfg_str) + self.LEN_OFFSET
+            p = struct.pack("!H", length) + p[2:]
+        return (p + pay)
+
+
+class SDOption_LoadBalance(_SDOption):
+    name = "LoadBalance Option"
+    _defaults = {'type': _SDOption.LOADBALANCE_TYPE,
+                 'len': _SDOption.LOADBALANCE_LEN}
+    fields_desc = [
+        _SDOption_Header,
+        ShortField("priority", 0),
+        ShortField("weight", 0)]
+
+
+class SDOption_IP4_EndPoint(_SDOption_IP4):
+    name = "IP4 EndPoint Option"
+    _defaults = {'type': _SDOption.IP4_ENDPOINT_TYPE,
+                 'len': _SDOption.IP4_ENDPOINT_LEN}
+
+
+class SDOption_IP4_Multicast(_SDOption_IP4):
+    name = "IP4 Multicast Option"
+    _defaults = {'type': _SDOption.IP4_MCAST_TYPE,
+                 'len': _SDOption.IP4_MCAST_LEN}
+
+
+class SDOption_IP4_SD_EndPoint(_SDOption_IP4):
+    name = "IP4 SDEndPoint Option"
+    _defaults = {'type': _SDOption.IP4_SDENDPOINT_TYPE,
+                 'len': _SDOption.IP4_SDENDPOINT_LEN}
+
+
+class SDOption_IP6_EndPoint(_SDOption_IP6):
+    name = "IP6 EndPoint Option"
+    _defaults = {'type': _SDOption.IP6_ENDPOINT_TYPE,
+                 'len': _SDOption.IP6_ENDPOINT_LEN}
+
+
+class SDOption_IP6_Multicast(_SDOption_IP6):
+    name = "IP6 Multicast Option"
+    _defaults = {'type': _SDOption.IP6_MCAST_TYPE,
+                 'len': _SDOption.IP6_MCAST_LEN}
+
+
+class SDOption_IP6_SD_EndPoint(_SDOption_IP6):
+    name = "IP6 SDEndPoint Option"
+    _defaults = {'type': _SDOption.IP6_SDENDPOINT_TYPE,
+                 'len': _SDOption.IP6_SDENDPOINT_LEN}
+
+
+##
+# SD PACKAGE DEFINITION
+##
+class SD(_SDPacketBase):
+    """
+    SD Packet
+
+    NOTE :   when adding 'entries' or 'options', do not use list.append()
+        method but create a new list
+    e.g. :  p = SD()
+            p.option_array = [SDOption_Config(),SDOption_IP6_EndPoint()]
+    """
+    SOMEIP_MSGID_SRVID = 0xffff
+    SOMEIP_MSGID_SUBID = 0x1
+    SOMEIP_MSGID_EVENTID = 0x100
+    SOMEIP_PROTO_VER = 0x01
+    SOMEIP_IFACE_VER = 0x01
+    SOMEIP_MSG_TYPE = SOMEIP.TYPE_NOTIFICATION
+
+    name = "SD"
+    _sdFlag = collections.namedtuple('Flag', 'mask offset')
+    FLAGSDEF = {
+        "REBOOT": _sdFlag(mask=0x80, offset=7),
+        "UNICAST": _sdFlag(mask=0x40, offset=6)
+    }
+
+    name = "SD"
+    fields_desc = [
+        ByteField("flags", 0),
+        X3BytesField("res", 0),
+        FieldLenField("len_entry_array", None,
+                      length_of="entry_array", fmt="!I"),
+        PacketListField("entry_array", None, cls=_SDEntry,
+                        length_from=lambda pkt: pkt.len_entry_array),
+        FieldLenField("len_option_array", None,
+                      length_of="option_array", fmt="!I"),
+        PacketListField("option_array", None, cls=_SDOption,
+                        length_from=lambda pkt: pkt.len_option_array)]
+
+    def getFlag(self, name):
```

Comment:
Can you use `get_flag`? (same for the other methods below)

Reply:
Done. Thanks for the review and the suggestions.